### PR TITLE
Parse: Change `rz_parse_subvar` API to accept a `RzAnalysisOp`

### PR DESCRIPTION
**Pull request by GustavoLCR**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Avoids reanalyzing the op, speeding up disassembly

**Test plan**

Tests pass

**Closing issues**

None

### CODE REVIEW

File: librz/parse/parse.c [hunk 1]
```diff
@@ -154,9 +154,9 @@ RZ_API char *rz_parse_immtrim(char *opstr) {
 	return opstr;
 }
 
-RZ_API bool rz_parse_subvar(RzParse *p, RzAnalysisFunction *f, ut64 addr, int oplen, char *data, char *str, int len) {
+RZ_API bool rz_parse_subvar(RzParse *p, RzAnalysisFunction *f, RzAnalysisOp *op, char *data, char *str, int len) {
```
Comment (XVilka) [inline librz/parse/parse.c]: Please add also a Doxygen

**Review by XVilka — APPROVED** [review]
Looks good

**Comment by XVilka** [comment]
`clang-format` failure: https://github.com/rizinorg/rizin/runs/6719864010?check_suite_focus=true#step:7:1553

File: librz/parse/parse.c [hunk 2]
```diff
@@ -154,9 +154,18 @@ RZ_API char *rz_parse_immtrim(char *opstr) {
...
+* \brief Substitutes register relative accesses with function variable names
+* \param p The parser
+* \param f The function
+* \param op The analysis op of the current instruction
+* \param data The disassembly of the current instruction
+* \param str The string buffer to write the output to
+* \param len The length of the string buffer
+*/
+RZ_API bool rz_parse_subvar(RZ_BORROW RZ_NONNULL RzParse *p, RZ_BORROW RZ_NULLABLE RzAnalysisFunction *f, RZ_BORROW RZ_NONNULL RzAnalysisOp *op, RZ_BORROW RZ_NONNULL RZ_IN char *data, RZ_BORROW RZ_NONNULL RZ_OUT char *str, int len) {
 	if (p->cur && p->cur->subvar) {
```
Comment (wargio) [inline librz/parse/parse.c]: add an rz_return_if_fail for each RZ_NONNULL, also i do not think is needed to specify `RZ_BORROW` when you pass a variable as input, jusr RZ_IN/RZ_OUT for pointers that are used to return a value (which in this case you do not have)

**Review by wargio — CHANGES_REQUESTED** [review]

File: librz/parse/parse.c (hunk 2 above)
Comment (XVilka) [reply librz/parse/parse.c]: Agree

**Review by wargio — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Pull request merged**